### ISSUE TITLE
Incorrect detection of setter for method name "setting()"

### ISSUE THREAD

**Issue by vijayvepa** [FasterXML/jackson-databind#2926]
**Describe the bug**
One of my methods is called `setting(MyClass value)` it is being detected as a setter and reported as a property `ting`.  

**Version information**
Which Jackson version(s) was this for? 
`Jackson-databind 2.9.2`

**To Reproduce**
If you have a way to reproduce this with:

1. Brief code sample/snippet: include here in preformatted/code section

I have a class with following method (generated by swagger-codegen)
```
  public SettingCreationRequestCM setting(SettingRequestCM setting) {
    this.setting = setting;
    return this;
  }

  /**
   * Get setting
   * @return setting
  **/
  @ApiModelProperty(required = true, value = "")
  @NotNull

  @Valid

  public SettingRequestCM getSetting() {
    return setting;
  }

  public void setSetting(SettingRequestCM setting) {
    this.setting = setting;
  }
```

I am using Spring-Fox to validate this class against below swagger YML snippet.

```yml

```

2. Longer example stored somewhere else (diff repo, snippet), add a link
3. Textual explanation: include here

**Expected behavior**
The method should be ignored and not treated as a setter

**Additional context**
This is failing my swagger test

**Possible Fix**

in /jackson-databind-2.9.9-sources.jar!/com/fasterxml/jackson/databind/util/BeanUtil.java

Current Method:

```java
    protected static String legacyManglePropertyName(final String basename, final int offset)
    {
        final int end = basename.length();
        if (end == offset) { // empty name, nope
            return null;
        }
        // next check: is the first character upper case? If not, return as is
        char c = basename.charAt(offset);
        char d = Character.toLowerCase(c);
        
        if (c == d) {
            return basename.substring(offset);
        }
        ...
    }
```

Proposed Fix:

```java
    protected static String legacyManglePropertyName(final String basename, final int offset)
    {
        final int end = basename.length();
        if (end == offset) { // empty name, nope
            return null;
        }
        // next check: is the first character upper case? If not, return as is
        char c = basename.charAt(offset);
        char d = Character.toLowerCase(c);
        
        if (c == d) { //non-setter method, nope
            return null;
        }
        ...
    }
```

**Comment by vijayvepa** [comment]
Debugger at the call stack 0

![image](https://user-images.githubusercontent.com/4973484/98571008-10ed1d80-2282-11eb-9693-e8ed573b6f8c.png)

Debugger at call stack 1

![image](https://user-images.githubusercontent.com/4973484/98571098-2bbf9200-2282-11eb-900f-6d6cd934d2f8.png)

Debugger at call stack 2

![image](https://user-images.githubusercontent.com/4973484/98571168-3e39cb80-2282-11eb-9fda-9a365cc8d272.png)

**Comment by cowtowncoder** [comment]
Yes, this is the current behavior: `set` and `get` do NOT have to be followed by a capital case letter. So it is not a bug per se.
It would be good to know if there is Java Beans specification that mandates capital case letter, which could be used to argue that there should be way to change behavior.

Jackson 2.12 will introduce a way to change this, however, issue #2800 introduces pluggable `AccessorNamingStrategy`, as well as default implementation that can be configured in a way to disallow methods like `setting()`.

**Comment by vijayvepa** [comment]
Related Pull Request: https://github.com/FasterXML/jackson-databind/pull/2930

**Comment by vijayvepa** [comment]
> Yes, this is the current behavior: `set` and `get` do NOT have to be followed by a capital case letter. So it is not a bug per se.
> It would be good to know if there is Java Beans specification that mandates capital case letter, which could be used to argue that there should be way to change behavior.
> 
> Jackson 2.12 will introduce a way to change this, however, issue #2800 introduces pluggable `AccessorNamingStrategy`, as well as default implementation that can be configured in a way to disallow methods like `setting()`.

- I noticed from existing tests that `setaField` is treated as a setter for `aField`

I experimented with field generation in IntelliJ, it does create `setaField` as the setter for `aField`, 
Also, it creates `setAnField` as a setter for `anField`. 

I have covered those cases in the PR

**Comment by cowtowncoder** [comment]
@vijayvepa I am not sure how IntelliJ's handling is relevant here. Jackson's logic for detecting properties does not start with fields (although they are taken into account) but instead considers accessors (setters, getters, fields) separate, then joining them.

But the fact is that this would change existing behavior for users, some of who may be relying on similar setters (more likely `set_score()` or something) getting detected.
Because of this, any changes would need to go in a minor version. 2.12 is being locked down now, so changes would need to go in 2.13.

I will leave this open so that issue can be considered for 2.12. Until then, I'd suggest users to see how to use #2800 (once 2.12 is released) -- and before that, to use annotations (for ignoring incorrectly detected property, `@JsonIgnore`) to indicate that specific accessor should not imply property.

**Comment by vijayvepa** [comment]
ok i will try JsonIgnore, thanks Tatu

**Comment by cowtowncoder** [comment]
I think this is duplicate of #2882, fixed for 3.0.0. Closing.
